src/modules/rover_ackermann/RoverAckermannGuidance/RoverAckermannGuidance.cpp

```diff
@@ -268,8 +268,10 @@ float RoverAckermannGuidance::updateAcceptanceRadius(const Vector2f &curr_wp_ned
 
 	// Calculate acceptance radius s.t. the rover cuts the corner tangential to the current and next line segment
 	if (curr_to_next_wp_ned.norm() > FLT_EPSILON && curr_to_prev_wp_ned.norm() > FLT_EPSILON) {
-		const float theta = acosf((curr_to_prev_wp_ned * curr_to_next_wp_ned) / (curr_to_prev_wp_ned.norm() *
-					  curr_to_next_wp_ned.norm())) / 2.f;
+		float cosin = (curr_to_prev_wp_ned * curr_to_next_wp_ned) / (curr_to_prev_wp_ned.norm() *
+				curr_to_next_wp_ned.norm());
+		cosin = math::constrain<float>(cosin, -1.f, 1.f); // Protect against float precision problem
+		const float theta = acosf(cosin) / 2.f;
```

Comment:
I guess it was because the norm was 0. You can simply use [unit_or_zero](https://github.com/PX4/PX4-Autopilot/blob/75ce550db39cc5d61ba970ca5f4b6096a287827e/src/lib/matrix/matrix/Vector.hpp#L132) that handles the division by zero for you.
```suggestion
		const float theta = acosf(curr_to_prev_wp_ned.unit_or_zero() * curr_to_next_wp_ned.unit_or_zero()) / 2.f;
```

Reply:
It's what I thought at the beginning too, but the norm wasn't zero.

It's because `(curr_to_prev_wp_ned * curr_to_next_wp_ned)` is really slightly bigger than `(curr_to_prev_wp_ned.norm() * curr_to_next_wp_ned.norm())`.

If the norm was 0, the value `cosin` I print in the video would be NaN of INF, but it's not, it's theta that is NaN, because the value given to `acosf` is `-1.000000119209290` so outside of the range [-1.0;1.0]

Reply:
I tried your solution but the problem still persist.
On the same mission that I show in the video I had several case where I had the following values :
`curr_to_prev_wp_ned.unit_or_zero() * curr_to_next_wp_ned.unit_or_zero() -1.000000119209290 | theta nan`

So it doesn't seems to resolve the problem of float precision.

Reply:
Okay, you convinced me! I will accept your solution but also add the protection against division-by-zero then